frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx

```diff
@@ -165,6 +235,238 @@ export const CreateSubscriptionYAML: React.SFC<CreateSubscriptionYAMLProps> = (p
+    return (
+      <ResourceLink
+        kind={referenceForModel(ConfigMapModel)}
+        name={catalogSource.spec.configmap}
+        namespace={catalogSource.metadata.namespace}
+      />
+    );
+  }
+  return catalogSource.spec.image || catalogSource.spec.address;
+};
+
+const getOperatorCount = (catalogSource, packageManifests) =>
+  _.filter(packageManifests, {
+    status: {
+      catalogSource: catalogSource.metadata.name,
+      catalogSourceNamespace: catalogSource.metadata.namespace,
+    },
+  }).length;
+
+const CatalogSourceTableRow: React.FC<CatalogSourceTableRowProps> = ({
+  obj: {
+    availability = '-',
+    disabled = false,
+    endpoint = '-',
+    isDefault = false,
+    name,
+    operatorCount = 0,
+    operatorHub,
+    publisher = '-',
+    source,
+  },
+  index,
+  key,
+  style,
+}) => {
+  return (
+    <TableRow
+      className={disabled && 'catalog-source__table-row--disabled'}
+      id={source ? source.metadata.uid : index}
+      index={index}
+      style={style}
+      trKey={key}
+    >
+      <TableData className={tableColumnClasses[0]}>
+        {source ? (
+          <ResourceLink
+            kind={catalogSourceModelReference}
+            name={source.metadata.name}
+            namespace={source.metadata.namespace}
+            title={source.metadata.name}
+          />
+        ) : (
+          name
+        )}
+      </TableData>
+      <TableData className={tableColumnClasses[1]}>{publisher || '-'}</TableData>
+      <TableData className={tableColumnClasses[2]}>{availability}</TableData>
+      <TableData className={tableColumnClasses[3]}>{endpoint || '-'}</TableData>
+      <TableData className={tableColumnClasses[4]}>{operatorCount || '-'}</TableData>
+      <TableData className={tableColumnClasses[5]}>
+        {isDefault ? (
+          <DefaultSourceKebab
+            kind={OperatorHubModel}
+            operatorHub={operatorHub}
+            sourceName={name}
+            sourceDisabled={disabled}
+          />
+        ) : (
+          <ResourceKebab
+            actions={[Kebab.factory.Edit, deleteModal]}
+            kind={catalogSourceModelReference}
+            resource={source}
+          />
+        )}
+      </TableData>
+    </TableRow>
+  );
+};
+
+const CatalogSourceList: React.FC<TableProps> = (props) => (
+  <Table
+    {...props}
+    aria-label="Catalog Sources"
+    Header={CatalogSourceHeader}
+    Row={CatalogSourceTableRow}
+  />
+);
+
+const flatten = ({
+  catalogSources,
+  operatorHub,
+  packageManifests,
+}: FlattenArgType): CatalogSourceTableRowObj[] => {
+  const defaultSources = _.map(operatorHub.status.sources, (defaultSource) => {
+    const catalogSource = _.find(catalogSources.data, {
+      metadata: { name: defaultSource.name, namespace: DEFAULT_SOURCE_NAMESPACE },
+    });
+    const catalogSourceExists = !_.isEmpty(catalogSource);
+    return {
+      availability: !catalogSourceExists ? 'Disabled' : 'Cluster wide',
```

Comment:
nit: easier to think about this way imo

```suggestion
      availability: catalogSourceExists ? 'Cluster wide' : 'Disabled',
```